src/bun.js/api/bun/h2_frame_parser.zig

```diff
@@ -0,0 +1,2440 @@
+            .{ "onEnd", "end" },
+            .{ "onError", "error" },
+            .{ "onGoAway", "goaway" },
+            .{ "onFrameError", "frameError" },
+            .{ "onAborted", "aborted" },
+            .{ "onWrite", "write" },
+        };
+
+        inline for (pairs) |pair| {
+            if (opts.getTruthy(globalObject, pair.@"1")) |callback_value| {
+                if (!callback_value.isCell() or !callback_value.isCallable(globalObject.vm())) {
+                    exception.* = JSC.toInvalidArguments(comptime std.fmt.comptimePrint("Expected \"{s}\" callback to be a function", .{pair.@"1"}), .{}, globalObject).asObjectRef();
+                    return null;
+                }
+
+                @field(handlers, pair.@"0") = callback_value;
+            }
+        }
+
+        if (handlers.onWrite == .zero) {
+            exception.* = JSC.toInvalidArguments("Expected at least \"write\" callback", .{}, globalObject).asObjectRef();
+            return null;
+        }
+
+        if (opts.getTruthy(globalObject, "binaryType")) |binary_type_value| {
+            if (!binary_type_value.isString()) {
+                exception.* = JSC.toInvalidArguments("Expected \"binaryType\" to be a string", .{}, globalObject).asObjectRef();
+                return null;
+            }
+
+            handlers.binary_type = BinaryType.fromJSValue(globalObject, binary_type_value) orelse {
+                exception.* = JSC.toInvalidArguments("Expected 'binaryType' to be 'arraybuffer', 'uint8array', 'buffer'", .{}, globalObject).asObjectRef();
+                return null;
+            };
+        }
+
+        return handlers;
+    }
+
+    pub fn unprotect(this: *Handlers) void {
+        this.onError.unprotect();
+        this.onGoAway.unprotect();
+        this.onWrite.unprotect();
+        this.onStreamError.unprotect();
+        this.onStreamStart.unprotect();
+        this.onStreamHeaders.unprotect();
+        this.onStreamEnd.unprotect();
+        this.onStreamData.unprotect();
+        this.onStreamError.unprotect();
+        this.onLocalSettings.unprotect();
+        this.onRemoteSettings.unprotect();
+        this.onWantTrailers.unprotect();
+        this.onPing.unprotect();
+        this.onEnd.unprotect();
+        this.onFrameError.unprotect();
+        this.onAborted.unprotect();
+    }
+
+    pub fn clear(this: *Handlers) void {
+        this.onError = .zero;
+        this.onWrite = .zero;
+        this.onStreamError = .zero;
+        this.onStreamStart = .zero;
+        this.onStreamHeaders = .zero;
+        this.onStreamEnd = .zero;
+        this.onStreamData = .zero;
+        this.onStreamError = .zero;
+        this.onLocalSettings = .zero;
+        this.onRemoteSettings = .zero;
+        this.onWantTrailers = .zero;
+        this.onPing = .zero;
+        this.onEnd = .zero;
+        this.onGoAway = .zero;
+        this.onFrameError = .zero;
+        this.onAborted = .zero;
+    }
+
+    pub fn protect(this: *Handlers) void {
```

Comment:
It doesn't feel great to be protecting this many values. It would be smarter to have a single JSInternalFieldsObjectImpl with all these values and then protecting that one value or using a `JSC.Strong`. 

But I don't think that should block this from shipping. Maybe a comment would be nice though.

Reply:
Actually, considering this is used in `H2FrameParser`, yes it should be a value.

Reply:
indeed, I started by using the socket.zig approach but a value should be better